src/Mealz/MealBundle/Controller/ParticipantController.php

```diff
@@ -6,19 +6,53 @@
 
 use App\Mealz\MealBundle\Entity\Day;
 use App\Mealz\MealBundle\Entity\DayRepository;
+use App\Mealz\MealBundle\Entity\Dish;
 use App\Mealz\MealBundle\Entity\Participant;
 use App\Mealz\MealBundle\Entity\Week;
 use App\Mealz\MealBundle\Entity\WeekRepository;
+use App\Mealz\MealBundle\Service\Exception\ParticipationException;
 use App\Mealz\MealBundle\Service\Notification\NotifierInterface;
+use App\Mealz\MealBundle\Service\ParticipationService;
 use App\Mealz\UserBundle\Entity\Profile;
 use DateTime;
+use Exception;
 use Sensio\Bundle\FrameworkExtraBundle\Configuration\Security;
 use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\Translation\Translator;
 
+/**
+ * @Security("is_granted('ROLE_USER')")
+ */
 class ParticipantController extends BaseController
 {
+    public function update(Request $request, Participant $participant, ParticipationService $participationSrv): JsonResponse
+    {
+        $dishSlugs = $request->request->get('dishes', []);
+
+        try {
+            $participationSrv->update($participant, $dishSlugs);
+        } catch (ParticipationException $pex) {
+            return new JsonResponse(['error' => $pex->getMessage()], 422);
+        } catch (Exception $exc) {
+            $this->logException($exc);
+
+            return new JsonResponse(['error' => 'unexpected error'], 500);
+        }
+
+        return new JsonResponse(
+            [
+                'actionText' => 'update',
```

Comment:
The actionText (at least for others) was always what happened. So it's in the past "updated", which means you get the response with status 200 and it has "updated" the participant.